source/slang/slang-reflection-api.cpp

```diff
@@ -1828,6 +1828,7 @@ namespace Slang
                     case LayoutResourceKind::ShaderResource:
                     case LayoutResourceKind::DescriptorTableSlot:
                     case LayoutResourceKind::Uniform:
+                    case LayoutResourceKind::ConstantBuffer: // for metal
```

Comment:
@csyonghe I'm not sure why this is needed, please take a look.

Reply:
All buffers (uav) maps to ConstantBuffer binding in metal. This means that a StructuredBuffer will have a ConstantBuffer resource layout, so this is correct.